subprojects/integ-test/src/integTest/groovy/org/gradle/integtests/TaskUpToDateIntegrationTest.groovy

```diff
@@ -50,22 +52,217 @@ class TaskUpToDateIntegrationTest extends AbstractIntegrationSpec {
         skippedTasks.contains ":myTask"
 
         when:
-        buildFile.text = buildFileWithOutputs "out2", "out1"
-        run "myTask"
+        run "myTask", "-Preverse"
         then:
-        skippedTasks.contains ":myTask"
+        executedAndNotSkipped ":myTask"
 
         where:
         annotation           | files
         '@OutputFiles'       | true
         '@OutputDirectories' | false
     }
 
-    def buildFileWithOutputs(String... outputs) {
-        """
+    @Issue("https://issues.gradle.org/browse/GRADLE-3540")
+    def "hash set output files do not make task out of date"() {
```

Comment:
Should we standardize on `out-of-date` or `not up-to-date`?

Reply:
I standardized on `not up-to-date`.